pages/api/callback.js

```diff
@@ -1,10 +1,18 @@
 import auth0 from '../../utils/auth';
+import { getPodName } from "../../utils/github"
+import Profile from "../../models/profile"
 
 export default async function callback(req, res) {
     try {
-        await auth0.handleCallback(req, res, { redirectTo: '/close' });
+        await auth0.handleCallback(req, res, {
+            redirectTo: '/close', onUserLoaded: async (req, res, session, state) => {
+                let fellow = await getPodName(session.user.nickname)
+                await Profile.create(session.user)
+                if (fellow.length == 0) throw new Error({ message: "It is only for mlh fellows" })
+                return session;
+            }
+        });
     } catch (error) {
-        console.error(error);
         res.status(error.status || 400).end(error.message);
```

Comment:
I think it might be better to return a status `500` instead of a `400` since it would be an unexpected error by this point.